CoreScriptsRoot/CoreScripts/Topbar.lua

```diff
@@ -2070,6 +2082,26 @@ else
 	topbarEnabledChanged()
 end
 
+if chatPrivacySettingEnabled then
+	spawn(function()
+		if Util.IsTouchDevice() or ChatModule:IsBubbleChatOnly() then
+			local success, localUserCanChat = pcall(function()
+				return ChatService:CanUserChatAsync(LocalPlayer.UserId)
```

Comment:
If the ChatService API goes down or fails this will disable chat for the user, is this the desired behavior? Could we retry if it fails?